Code/Tools/AssetProcessor/native/ui/ProductDependencyTreeItemData.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include "ProductDependencyTreeItemData.h"
+
+#include <AzCore/std/smart_ptr/make_shared.h>
+
+AZ_PUSH_DISABLE_WARNING(4127 4251 4800 4244, "-Wunknown-warning-option")
```

Comment:
I ported this code from your PR #6487, and I did not notice this line. Should I remove this line and check if the program still works properly?

Reply:
You could try removing it, but I think that's necessary. Without it, including certain Qt headers will cause those warnings to occur. You can search AZ_PUSH_DISABLE_WARNING in the codebase and find other use cases that might describe each individual warning. I only brought it up because it seemed like a good C++ gotcha to figure out here. I don't think we need the comment, but it's useful to understand why these warnings have to be temporarily disabled like this.

Reply:
actually, it compiled and ran on my side, after removing `AZ_PUSH_DISABLE_WARNING` and `AZ_POP_DISABLE_WARNING` in both `ProductAssetTreeItemData.cpp` and `ProductDependencyTreeItemData.cpp`. I will remove them and make a commit.

Reply:
finished